java/vector/src/main/java/org/apache/arrow/vector/stream/ArrowStreamReader.java

```diff
@@ -36,6 +38,7 @@
   private ReadChannel in;
   private final BufferAllocator allocator;
   private Schema schema;
```

Comment:
schema is read from the input stream now, so can't be final as it's not initially loaded